dataprofiler/tests/profilers/test_categorical_column_profile.py

```diff
@@ -308,6 +308,47 @@ def test_gini_impurity(self):
         profile = CategoricalColumn(df_categorical.name)
         profile.update(df_categorical)
         self.assertEqual(profile.gini_impurity, None)
+        
+    def test_categorical_diff(self):
+        df_categorical = pd.Series(["y", "y", "y", "y", "n", "n", "n"])
+        profile = CategoricalColumn(df_categorical.name)
+        profile.update(df_categorical)
+
+        df_categorical = pd.Series(["y", "maybe", "y", "y", "n", "n", "maybe"])
+        profile2 = CategoricalColumn(df_categorical.name)
+        profile2.update(df_categorical)
+        
+        expected_diff = {
+            'categorical': 'unchanged',
+            'statistics': {
+                'unique_count': -1,
+                'unique_ratio': -0.14285714285714285,
+                'categories': [[], ['y', 'n'], ['maybe']],
+                'gini_impurity': -0.16326530612244894,
+                'categorical_count': {
+                    'y': 1,
+                    'n': 1,
+                    'maybe': [None, 2]
+                }
+            }
+        }
+
+        self.assertDictEqual(expected_diff, profile.diff(profile2))
+
+        # Test with one categorical column matching
+        df_categorical = pd.Series(["THIS", "is", "not", "a", "categorical", 
```

Comment:
If this is supposed to not be categorical, naming it `df_categorical` is a bit confusing.

Reply:
haha good catch